src/gui/uithememanager.cpp

```diff
@@ -140,3 +154,81 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
 #endif
     return IconProvider::instance()->getIconPath(iconId);
 }
+
+void UIThemeManager::loadColorsFromJSONConfig()
+{
+    const QString configFileName(":uitheme/config.json");
+    QFile configFile(configFileName);
+    if (!configFile.open(QIODevice::ReadOnly)) {
+        LogMsg(tr("Failed to open \"%1\". Reason: %2").arg(configFileName, configFile.errorString()), Log::WARNING);
+        return;
+    }
+
+    QJsonParseError jsonError;
+    const QJsonDocument configJsonDoc = QJsonDocument::fromJson(configFile.readAll(), &jsonError);
+    if (jsonError.error != QJsonParseError::NoError) {
+        LogMsg(tr("Invalid JSON provided in \"%1\". Reason: %2").arg(configFileName, jsonError.errorString()), Log::WARNING);
+        return;
+    }
+    if (!configJsonDoc.isObject()) {
+        LogMsg(tr("Invalid JSON provided in \"%1\". Reason: %2").arg(configFileName, tr("Root JSON value is not an object")), Log::WARNING);
+        return;
+    }
+
+    const QJsonObject colors = configJsonDoc.object().value("colors").toObject();
+    for (auto color = colors.constBegin(); color != colors.constEnd(); ++color) {
+        QColor providedColor(color.value().toString());
+        if (!providedColor.isValid()) {
+            LogMsg(tr("Invalid color for ID \"%1\" is provided by theme").arg(color.key()));
```

Comment:
Maybe `Theme file provided invalid color for ID \"%1\"`?
And this should be `Log::WARNING` level.

Reply:
>Maybe `Theme file provided invalid color for ID \"%1\"`?

Do I have to use the past tense here?

Reply:
I think mentioning the file name is important, otherwise users would not know which file has problem.
`ID \"%1\" has invalid color value: \"%2\". Theme file: %3`

Reply:
>I think mentioning the file name is important

What file do you mean?
>otherwise users would not know which file has problem.

The file that is currently configured.

Reply:
IMO, we should provide info about currently used theme file only once we start to load it.

Reply:
>What file do you mean?

Of course it is: ":uitheme/config.json".

>The file that is currently configured.

Unless you look into code otherwise a user won't know that. A verbose warning message could help users and theme creators to track down errors easily.

Reply:
>Of course it is:  ":uitheme/config.json".

Sorry, you confuses me by wrong terminology. When we say "Theme file" we should always mean ". qbtheme" file.
>Unless you look into code otherwise a user won't know that.

Why should user know something about internal layout of theme file?
Theme creator should know it purposely (e.g. from some manuals). Otherwise how it can create themes?

I really dislike such over-verbose logging. Often, you provide too many unnecessary details, but you forget about the really important ones.

Reply:
Let's try to look at these messages from the user/theme creator perspective (i.e. when it see them in the log). This will help us understand what needs to be included.

It would also be nice to have some consistent style for all messages coming from the same component.

Reply:
any changes for here?

Reply:
>Sorry, you confuses me by wrong terminology. When we say "Theme file" we should always mean ". qbtheme" file.

But a .qbtheme file is a bundle, it doesn't matter to normal user but it matters to theme creators.

>Why should user know something about internal layout of theme file?

Usually user don't. But for theme creators, unless you have a validator program for .qbtheme file such error can only be diagnosed in qbt.

>I really dislike such over-verbose logging. Often, you provide too many unnecessary details, but you forget about the really important ones.

I disagree it applies to the case here.

>Let's try to look at these messages from the user/theme creator perspective (i.e. when it see them in the log). This will help us understand what needs to be included.

@jagannatharjun is a theme creator, we can ask his opinion.
What do you think about the log messages I suggested in https://github.com/qbittorrent/qBittorrent/pull/12719#discussion_r419243871 and https://github.com/qbittorrent/qBittorrent/pull/12719#discussion_r419283093? Does it help to clarify where the error message come from? (IMO the message in PR is a bit weird to read)

Reply:
>It would also be nice to have some consistent style for all messages coming from the same component.

Regardless of other stuff, this sentence still matters. An ordered and consistent log contributes to better diagnostics.

Reply:
personally speaking theme file name inside `ID \"%1\" has invalid color value: \"%2\". Theme file: %3` isn't actually more helpful from `Invalid color for ID \"%1\" is provided by theme` 

1. End-user doesn't know about `uitheme/config.json`. it will just create confusion Users will think "is it .qbtheme or this new weird-looking path `:uitheme/config.json`, which one of these two is handling my colors".

2. It may be useful for theme creators, but they already should know `uitheme/config.json` handles colors, this additional information will not help here either, 

huh but if we show some log message if the colors are empty (no colors are provided), then it may be helpful

Again this is my personal opinion and not so strong one

Reply:
ok. Let's keep it as is.

Reply:
Let's move on. Better logging can be provided in the future.